src/install/lockfile.zig

```diff
@@ -1599,48 +1582,18 @@ pub fn getOrPutID(this: *Lockfile, id: PackageID, name_hash: PackageNameHash) !v
         this.unique_packages.unset(id);
 
         switch (index.*) {
-            .PackageID => |single_| {
-                var ids = try this.allocator.alloc(PackageID, 8);
-                std.mem.set(PackageID, ids, invalid_package_id - 1);
-
-                ids[0] = single_;
-                ids[1] = id;
-                this.unique_packages.unset(single_);
+            .PackageID => |single| {
+                this.unique_packages.unset(single);
 
-                // stage1 compiler doesn't like this
-                ids[7] = invalid_package_id;
-                var ids_sentinel = ids.ptr[0 .. ids.len - 1 :invalid_package_id];
+                var ids = try PackageIDList.initCapacity(this.allocator, 8);
+                ids.appendAssumeCapacity(single);
+                ids.appendAssumeCapacity(id);
                 index.* = .{
-                    .PackageIDMultiple = ids_sentinel,
+                    .PackageIDMultiple = ids,
                 };
             },
-            .PackageIDMultiple => |ids_| {
-                var ids = bun.sliceTo(ids_, invalid_package_id - 1);
-                for (ids) |id2, i| {
-                    if (id2 == invalid_package_id - 1) {
-                        ids[i] = id;
-                        return;
-                    }
-                }
-
-                var new_ids = try this.allocator.alloc(PackageID, ids.len + 8);
-                std.mem.set(PackageID, new_ids, invalid_package_id - 1);
-
-                defer this.allocator.free(ids);
-                for (ids) |id2, i| {
-                    new_ids[i] = id2;
-                }
-                new_ids[ids.len - 1] = id;
-                for (new_ids[ids.len .. new_ids.len - 2]) |_, i| {
-                    new_ids[i + ids.len] = invalid_package_id - 1;
-                }
-
-                // stage1 compiler doesn't like this
-                new_ids[new_ids.len - 1] = invalid_package_id;
-                var new_ids_sentinel = new_ids.ptr[0 .. new_ids.len - 1 :invalid_package_id];
-                index.* = .{
-                    .PackageIDMultiple = new_ids_sentinel,
-                };
+            .PackageIDMultiple => |*ids| {
```

Comment:
and this references the deferenced pointer

this will only set `len` in the stack copy, not the one inside the array

instead, we will need to use std.meta.activeTag(entry) and reference by foo.PacakgeIDMultiple

Reply:
Fixed in https://github.com/oven-sh/bun/pull/1970/commits/8043296c3a64b568dce865d392927d630c03c0d5